components/canvas/raqote_backend.rs

```diff
@@ -268,33 +266,23 @@ impl PatternHelpers for Pattern<'_> {
         }
     }
 
-    fn draw_rect(&self, rect: &Rect<f32>) -> Rect<f32> {
```

Comment:
Here, this one worked only on fill_rect, by making target rect smaller to not go in no-repeat zone of image.